plugins/swagger-apibody/src/index.ts

```diff
@@ -27,39 +31,45 @@ class SwaggerApiBody implements AmplicationPlugin {
     eventParams: CreateEntityControllerBaseParams
   ) {
     const { templateMapping, template, controllerBaseId } = eventParams;
+    try {
+      interpolate(template, templateMapping);
+      const classDeclaration = getClassDeclarationById(
+        template,
+        controllerBaseId
+      );
 
-    const classDeclaration = getClassDeclarationById(
-      template,
-      controllerBaseId
-    );
+      Object.keys(funcMethodMap).forEach((funcName) => {
+        const methodId = templateMapping[funcName] as namedTypes.Identifier;
+        const classMethod = getClassMethodById(classDeclaration, methodId);
 
-    Object.keys(funcMethodMap).forEach((funcName) => {
-      const methodId = templateMapping[funcName] as namedTypes.Identifier;
-      const classMethod = getClassMethodById(classDeclaration, methodId);
+        const currDecorator = builders.decorator(
+          builders.callExpression(
+            builders.memberExpression(
+              builders.identifier("swagger"),
+              builders.identifier("ApiBody")
+            ),
+            [
+              builders.objectExpression([
+                builders.objectProperty(
+                  builders.identifier("type"),
+                  builders.identifier(
+                    funcMethodMap[funcName as keyof typeof funcMethodMap]
+                  )
+                ),
+              ]),
+            ]
+          )
+        );
 
-      const currDecorator = builders.decorator(
-        builders.callExpression(
-          builders.memberExpression(
-            builders.identifier("swagger"),
-            builders.identifier("ApiBody")
-          ),
-          [
-            builders.objectExpression([
-              builders.objectProperty(
-                builders.identifier("type"),
-                builders.identifier(
-                  funcMethodMap[funcName as keyof typeof funcMethodMap]
-                )
-              ),
-            ]),
-          ]
-        )
-      );
+        classMethod?.decorators?.push(currDecorator);
+      });
 
-      classMethod?.decorators?.push(currDecorator);
-    });
+      return eventParams;
+    } catch (error) {
+      console.error("Failed to invoke beforeCreateControllerBase event under swagger-api-body plugin");
```

Comment:
Where will we be able to see this console.error when it's running inside the DSG container? is there a way to pass back the error to the build log?

Reply:
Let's use context.logger to get that message to the user. Plus we will also get it on our logging system.

Reply:
good catch... I printed it to context.logger